--- a/Makefile
+++ b/Makefile
@@ -1,2 +1,52 @@
 default %:
 	make $@ -C src
+
+default: doc/develop.pdf doc/example-c6h12.pdf doc/example-cu.pdf doc/formalism.pdf doc/inputformat.pdf doc/programs.pdf doc/tutorial.pdf
+
+distclean clean: clean-doc
+
+clean-doc:
+	rm -f doc/*-u.*
+	rm -f doc/develop.pdf doc/example-c6h12.pdf doc/example-cu.pdf doc/formalism.pdf doc/inputformat.pdf doc/programs.pdf doc/tutorial.pdf
+
+doc/develop.pdf:
+	(cd doc && nkf -u develop.tex > develop-u.tex)
+	(cd doc && platex develop-u.tex)
+	(cd doc && platex develop-u.tex)
+	(cd doc && dvipdfmx -o develop.pdf develop-u.dvi)
+
+doc/example-c6h12.pdf:
+	(cd doc && nkf -u example-c6h12.tex > example-c6h12-u.tex)
+	(cd doc && platex example-c6h12-u.tex)
+	(cd doc && platex example-c6h12-u.tex)
+	(cd doc && dvipdfmx -o example-c6h12.pdf example-c6h12-u.dvi)
+
+doc/example-cu.pdf:
+	(cd doc && nkf -u example-cu.tex > example-cu-u.tex)
+	(cd doc && platex example-cu-u.tex)
+	(cd doc && platex example-cu-u.tex)
+	(cd doc && dvipdfmx -o example-cu.pdf example-cu-u.dvi)
+
+doc/formalism.pdf:
+	(cd doc && nkf -u formalism.tex > formalism-u.tex)
+	(cd doc && platex formalism-u.tex)
+	(cd doc && platex formalism-u.tex)
+	(cd doc && dvipdfmx -o formalism.pdf formalism-u.dvi)
+
+doc/inputformat.pdf:
+	(cd doc && nkf -u inputformat.tex > inputformat-u.tex)
+	(cd doc && platex inputformat-u.tex)
+	(cd doc && platex inputformat-u.tex)
+	(cd doc && dvipdfmx -o inputformat.pdf inputformat-u.dvi)
+
+doc/programs.pdf:
+	(cd doc && nkf -u programs.tex > programs-u.tex)
+	(cd doc && platex programs-u.tex)
+	(cd doc && platex programs-u.tex)
+	(cd doc && dvipdfmx -o programs.pdf programs-u.dvi)
+
+doc/tutorial.pdf:
+	(cd doc && nkf -u tutorial.tex > tutorial-u.tex)
+	(cd doc && platex tutorial-u.tex)
+	(cd doc && platex tutorial-u.tex)
+	(cd doc && dvipdfmx -o tutorial.pdf tutorial-u.dvi)
