--- a/makefile
+++ b/makefile
@@ -3,15 +3,17 @@
 # The corresponding compiler option also should be given 
 # by "flag= ".
 #fort = f77
-#fort = gfortran
+fort = gfortran
 #flag = -mp1 -prec_div -tpp7 -axW -xW -mcmodel=medium -i-dynamic
 #.PHONY: all
 #all: $(program) gpd
-fort = ifort
+#fort = ifort
 flag = -O3
 libs =
 
 program = specx
+utility = fmg
+prefix=$(DESTDIR)/usr
 objs = \
 source/$(program).o \
 source/spmain.o \
@@ -206,10 +208,23 @@
 source/xtoken.o \
 source/zroots.o
 
+default: $(program) $(utility) gpd
 $(program): $(objs)
-	@$(fort) -o $(program) $(flag) $(objs) $(libs)
+	$(fort) -o $(program) $(flag) $(objs) $(libs)
+$(utility): util/fmg.f
+	$(fort) -o $(utility) $(flag) util/fmg.f $(libs)
 .f.o:
 	$(fort) $(flag) -o $(<:.f=.o) -c $<
 
 gpd: source/gpd.f
 	$(fort) $(flag) -o gpd source/gpd.f
+
+install: $(program) $(utility) gpd
+	mkdir -p $(prefix)/bin $(prefix)/share/akaikkr/in $(prefix)/share/akaikkr/util
+	cp $(program) $(utility) gpd $(prefix)/bin
+	chmod 755 $(prefix)/bin/$(program) $(prefix)/share/akaikkr/in $(prefix)/share/akaikkr/util
+	cp -p in/* $(prefix)/share/akaikkr/in
+	cp -p util/* $(prefix)/share/akaikkr/util
+	chmod 644 $(prefix)/share/akaikkr/in/* $(prefix)/share/akaikkr/util/*
+clean:
+	rm -f $(program) $(utility) gpd source/*.o
